lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -18,19 +18,49 @@
  */
 package org.apache.lens.cube.parse;
 
+import static org.apache.lens.cube.parse.CandidateTablePruneCause.*;
+import static org.apache.lens.cube.parse.StorageUtil.*;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
 import java.util.*;
 
 import org.apache.lens.cube.metadata.*;
+import org.apache.lens.server.api.error.LensException;
+import org.apache.lens.server.api.metastore.DataCompletenessChecker;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.util.ReflectionUtils;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
 
+import com.google.common.collect.Sets;
 import lombok.Getter;
 import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * Represents a fact on a storage table and the dimensions it needs to be joined with to answer the query
- *
  */
-public class StorageCandidate implements Candidate,CandidateTable {
+@Slf4j
+public class StorageCandidate implements Candidate, CandidateTable {
 
+  @Getter
+  private final CubeQueryContext cubeql;
```

Comment:
Can you comment on this member variables ? 
Also can we think of more apt names